tests/show_techsupport/test_techsupport.py

```diff
@@ -0,0 +1,253 @@
+    acl_rule_file = os.path.join(mirror_setup['dut_tmp_dir'], ACL_RULE_PERSISTENT_FILE)
+    extra_vars = {
+        'acl_table_name':  EVERFLOW_TABLE_NAME,
+    }
+    logger.info('extra variables for MIRROR table:\n{}'.format(pprint.pformat(extra_vars)))
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+
+    duthost.template(src=os.path.join(TEMPLATE_DIR, ACL_RULE_PERSISTENT_J2), dest=acl_rule_file)
+    duthost.command('config mirror_session add {} {} {} {} {} {} {}'
+    .format(SESSION_INFO['name'], SESSION_INFO['src_ip'], neighbor_ip,
+     SESSION_INFO['dscp'], SESSION_INFO['ttl'], SESSION_INFO['gre'], SESSION_INFO['queue']))
+
+    logger.info('Loading acl mirror rules ...')
+    load_rule_cmd = "acl-loader update full {} --session_name={}".format(acl_rule_file, SESSION_INFO['name'])
+    duthost.command('{}'.format(load_rule_cmd))
+
+    try:
+        yield
+    finally:
+        loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix='acl')
+        loganalyzer.load_common_config()
+
+        try:
+            loganalyzer.expect_regex = [LOG_EXCEPT_MIRROR_SESSION_REMOVE]
+            with loganalyzer:
+                teardown_mirroring(duthost, mirror_setup['dut_tmp_dir'])
+        except LogAnalyzerError as err:
+            raise err
+
+
+def teardown_mirroring(dut, tmp_path):
+    """
+    teardown EVERFLOW rules after test by applying empty configuration
+    :param dut: DUT host object
+    :param setup: setup information
+    :return:
+    """
+    logger.info('removing MIRRORING rules')
+    # copy rules remove configuration
+    dst = os.path.join(tmp_path, ACL_RULE_PERSISTENT_DEL_FILE)
+    dut.copy(src=os.path.join(FILES_DIR, ACL_RULE_PERSISTENT_DEL_FILE), dest=dst)
+    dut.command("acl-loader update full {}".format(dst))
+    dut.command('config mirror_session remove {}'.format(SESSION_INFO['name']))
+
+
+@pytest.fixture(scope='function', params=['acl', 'mirroring'], autouse=True)
+def config(request):
+    """
+    fixture to add configurations on setup by received parameters.
+    """
+    return request.getfixturevalue(request.param)
+
+
+def test_techsupport(request, config, duthost, testbed):
+    """
+    test the "show techsupport" command in a loop
+    :param config: fixture to configure additional setups_list on dut.
+    :param duthost: dut host
+    :param testbed: testbed
+    """
+    loop_range = request.config.getoption("--loop_num") or DEFAULT_LOOP_RANGE
+    loop_delay = request.config.getoption("--loop_delay") or DEFAULT_LOOP_DELAY
+    since = request.config.getoption("--logs_since") or randint(1, 60)
+
+    logger.debug("loop range is {} and loop delay is {}".format(loop_range, loop_delay))
+
+    for i in range(loop_range):
+        logger.debug("Running show techsupport ... ")
+        duthost.command("show techsupport --since='{} minute ago'".format(since))
+        logger.debug("Sleeping for {} seconds".format(loop_delay))
+        time.sleep(loop_delay)
+
```

Comment:
Few general comments. Once the test is completed we might have X results of 'show techsupport' executions. 
1. Should we should verify all are available on switch? 
2. Should we validate how much time it takes to run the command and in case it is more than X minutes we should raise error? In the case of logs_since which is assumed only minutes we must have it ready in minutes. and if not this should be treated as an error.
3. Should we remove all the created files (only those that were created) at the end of the test to make sure we don't take too much test of the disk just because of the test?